src/ray/gcs/store_client/redis_store_client.cc

```diff
@@ -415,26 +358,23 @@ void RedisStoreClient::RedisScanner::Scan(const std::string &match_pattern,
   size_t batch_count = RayConfig::instance().maximum_gcs_storage_operation_batch_size();
   ++pending_request_count_;
 
-  auto scan_callback =
-      [this, match_pattern, callback](const std::shared_ptr<CallbackReply> &reply) {
-        OnScanCallback(match_pattern, reply, callback);
-      };
+  auto scan_callback = [this, callback](const std::shared_ptr<CallbackReply> &reply) {
+    OnScanCallback(reply, callback);
+  };
   // Scan by prefix from Redis.
-  std::vector<std::string> args = {"HSCAN",
-                                   external_storage_namespace_,
-                                   std::to_string(cursor_.value()),
-                                   "MATCH",
-                                   match_pattern,
-                                   "COUNT",
-                                   std::to_string(batch_count)};
+  std::vector<std::string> args = {"HSCAN", table_name_, std::to_string(cursor_.value())};
```

Comment:
why does it not use GenRedisKey?

Reply:
Ctor of RedisScanner accepts the table name, and all its callers uses GenRedisKey so we are fine.

Reply:
Do you want a more typed set up, e.g. a struct represending the {external_storage_namespace, table_name} ?

Reply:
> Do you want a more typed set up, e.g. a struct represending the {external_storage_namespace, table_name} ?

Yeah this would be great. I am also confused why this doesn't include `external_storage_namespace` (that's why I was asking why we don't use GenRedisKey)?